ocaml/xapi-cli-server/records.ml

```diff
@@ -5252,3 +5252,159 @@ let vtpm_record rpc session_id vtpm =
           ()
       ]
   }
+
+let observer_record rpc session_id observer =
+  let _ref = ref observer in
+  let empty_record =
+    ToGet (fun () -> Client.Observer.get_record ~rpc ~session_id ~self:!_ref)
+  in
+  let record = ref empty_record in
+  let x () = lzy_get record in
+  {
+    setref=
+      (fun r ->
+        _ref := r ;
+        record := empty_record
+      )
+  ; setrefrec=
+      (fun (a, b) ->
+        _ref := a ;
+        record := Got b
+      )
+  ; record= x
+  ; getref= (fun () -> !_ref)
+  ; fields=
+      [
+        make_field ~name:"uuid" ~get:(fun () -> (x ()).API.observer_uuid) ()
+      ; make_field ~name:"host-uuids"
+          ~get:(fun () ->
+            map_and_concat get_uuid_from_ref (x ()).API.observer_hosts
+          )
+          ~set:(fun s ->
+            let value =
+              get_words ',' s
+              |> List.map (fun uuid ->
+                     Client.Host.get_by_uuid ~rpc ~session_id ~uuid
+                 )
+            in
+            Client.Observer.set_hosts ~rpc ~session_id ~self:observer ~value
+          )
+          ~remove_from_set:(fun uuid ->
+            let host = Client.Host.get_by_uuid ~rpc ~session_id ~uuid in
+            let hosts =
+              Client.Observer.get_hosts ~rpc ~session_id ~self:observer
+            in
+            let value = List.filter (( <> ) host) hosts in
+            Client.Observer.set_hosts ~rpc ~session_id ~self:observer ~value
```

Comment:
Shouldn't this be a single call to `Client.Observer.register_host` ?

Reply:
register and unregister do not have and DB side effects, in the case of hosts, a diff between hosts that need to be removed and those that need to be added needs to be established.

Reply:
I see your point now, but the DB would not be modified by just calling `unregister`